lib/gcloud/resource_manager/project.rb

```diff
@@ -0,0 +1,379 @@
+      # <code>[a-z]([-a-z0-9]*[a-z0-9])?</code>.
+      #
+      # Label values must be between 0 and 63 characters long and must conform
+      # to the regular expression <code>([a-z]([-a-z0-9]*[a-z0-9])?)?</code>.
+      #
+      # No more than 256 labels can be associated with a given resource.
+      #
+      # Clients should store labels in a representation such as JSON that does
+      # not depend on specific characters being disallowed.
+      #
+      # === Example
+      #
+      #   require "gcloud"
+      #
+      #   gcloud = Gcloud.new
+      #   resource_manager = gcloud.resource_manager
+      #   project = resource_manager.project "tokyo-rain-123"
+      #   project.labels = { "env" => "production" }
+      #
+      def labels= new_labels
+        ensure_connection!
+        @gapi["labels"] = new_labels
+        resp = connection.update_project @gapi
+        if resp.success?
+          @gapi = resp.data
+        else
+          fail ApiError.from_response(resp)
+        end
+      end
+
+      ##
+      # The time that this project was created.
+      #
+      def created_at
+        Time.parse @gapi["createTime"]
+      rescue
+        nil
+      end
+
+      ##
+      # The project lifecycle state.
+      #
+      # Acceptable values are:
+      # * +ACTIVE+ - The normal and active state.
+      # * +LIFECYCLE_STATE_UNSPECIFIED+ - Unspecified state. This is only
+      # used/useful for distinguishing unset values.
+      # * +DELETE_REQUESTED+ - The project has been marked for deletion by the
+      # user (by invoking DeleteProject) or by the system (Google Cloud
+      # Platform). This can generally be reversed by invoking UndeleteProject.
+      # * +DELETE_IN_PROGRESS+ - The process of deleting the project has begun.
+      # Reversing the deletion is no longer possible.
+      #
+      def state
+        @gapi["lifecycleState"]
+      end
+
+      ##
+      # Checks if the state is +ACTIVE+.
+      def active?
+        return false if state.nil?
+        "ACTIVE".casecmp(state).zero?
+      end
+
+      ##
+      # Checks if the state is +LIFECYCLE_STATE_UNSPECIFIED+.
+      def unspecified?
+        return false if state.nil?
+        "LIFECYCLE_STATE_UNSPECIFIED".casecmp(state).zero?
+      end
+
+      ##
+      # Checks if the state is +DELETE_REQUESTED+.
+      def delete_requested?
+        return false if state.nil?
+        "DELETE_REQUESTED".casecmp(state).zero?
+      end
+
+      ##
+      # Checks if the state is +DELETE_IN_PROGRESS+.
+      def delete_in_progress?
+        return false if state.nil?
+        "DELETE_IN_PROGRESS".casecmp(state).zero?
+      end
+
+      ##
+      # Updates the project in a single API call. See Project::Updater
+      #
+      # === Example
+      #
+      #   require "gcloud"
+      #
+      #   gcloud = Gcloud.new
+      #   resource_manager = gcloud.resource_manager
+      #   project = resource_manager.project "tokyo-rain-123"
+      #   project.update do |tx|
```

Comment:
Maybe change `tx` to `p`? We used `tx` in DNS because the official docs actually discuss a "transaction", but not sure if it's the correct choice for all update block examples. Just a thought...